supplier/models.py

```diff
@@ -0,0 +1,28 @@
+from django.db import models
+
+
+class User(models.Model):
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False)
+    first_name = models.CharField(max_length=32)
+    last_name = models.CharField(max_length=32)
+    password = models.CharField(max_length=16)
+    user_type = models.CharField(max_length=16)
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    business_name = models.CharField(max_length=32)
+
+    @staticmethod
+    def filter_by_field(field, field_value):
```

Comment:
If we filter by username its either a supplier or None
what if we filter by businessname it is either Supplier list or Empty list and not None

Reply:
If we try to filter by X and X is not in the supplier attributes we get None why not exception

Reply:
> If we filter by username its either a supplier or None what if we filter by businessname it is either Supplier list or Empty list and not None

in both cases i receive query set, not sure i understood what you meant.
@thetaxas2468

Reply:
Either you get query set or None here,what will happend if you typed wrong field that doesnt exist

Reply:
> Either you get query set or None here,what will happend if you typed wrong field that doesnt exist

FieldDoesNotExists error

Reply:
I guess you need to be aware from the exception in the tests with context manager like what i have done

Reply:
yea, ill change it to context manager.